core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2011,6 +2011,31 @@ class SparkContext(config: SparkConf) extends Logging {
       func: (TaskContext, Iterator[T]) => U,
       partitions: Seq[Int],
       resultHandler: (Int, U) => Unit): Unit = {
+    runJob[T, U](rdd, (_: Int) => {}, func, partitions, resultHandler)
+  }
+
+  /**
+   * Create configs on driver as part of commit protocol using a provided function that accepts
+   * an integer (stageId) as parameter and a function on a given set of partitions in an RDD and
+   * pass the results to the given handler function. This is the main entry point for all actions
+   * in Spark.
```

Comment:
I'd simplify this comment to something like

This is a private version of `runJob` which adds `jobSetupFunc`, just for SparkHadoopWriter, not for general users.